File: py/1029.py
def fib(n):
    f = [0, 1]

    for i in range(2, n + 1):
        f.append(f[i - 1] + f[i -2])

    return f


n = int(input())
fibs = fib(41)

for _ in range(n):
    i = int(input())
    print(f'fib({i}) = {2 * fibs[i + 1] - 2} calls = {fibs[i]}')

File: cpp/1029.cpp
#include <iostream>

using namespace std;

int main()
{
    int fibs[41];
    fibs[0] = 0;
    fibs[1] = 1;

    for (int i = 2; i < 42; i++) {
        fibs[i] = fibs[i - 1] + fibs[i - 2];
    }

    int n, m;
    cin >> n;

    for (n; n > 0; n--) {
        cin >> m;
        cout << "fib(" << m << ") = " << 2 * fibs[m + 1] - 2 << " calls = " << fibs[m] << '\n';
    }

    return 0;
}